lecli/log_api.py

```diff
@@ -0,0 +1,174 @@
+"""
+Log API module.
+"""
+import json
+import requests
+
+from lecli import api_utils
+from lecli import response_utils
+
+def _url():
+    """
+    Get rest query url of log resource id.
+    """
+    return 'https://rest.logentries.com/management/logs'
+
+
+def handle_get_log_response(response):
+    """
+    Handle get log response.
+    """
+    if response_utils.response_error(response):
+        exit(1)
+    elif response.status_code == 200:
+        if 'log' in response.json():
+            print json.dumps(response.json()['log'], indent=4, sort_keys=True)
+        elif 'logs' in response.json():
+            print json.dumps(response.json()['logs'], indent=4, sort_keys=True)
+
+
+def get_logs():
+    """
+    Get logs associated with the user.
+    """
+    headers = api_utils.generate_headers('ro')
+    try:
+        response = requests.request('GET', _url(), headers=headers)
+        handle_get_log_response(response)
+    except requests.exceptions.RequestException as error:
+        print error
+        exit(1)
+
+
+def get_log(log_id):
+    """
+    Get a specific log.
+    """
+    headers = api_utils.generate_headers('ro')
+    try:
+        response = requests.get(_url() + "/" + str(log_id),
+                                headers=headers)
+        handle_get_log_response(response)
+    except requests.exceptions.RequestException as error:
+        print error
+        exit(1)
+
+
+def create_log(logname, params):
+    """
+    Add a new log to the current account.
+    If a filename is given, load the contents of the file
+    as json parameters for the request.
+    If a name is given, create a new log with the given name
+    """
+
+    if params is not None:
+        request_params = params
+    else:
+        request_params = {
+            'log': {
+                'name': str(logname)
+            }
+        }
+
+    headers = api_utils.generate_headers('rw')
+
+    try:
+        response = requests.post(_url(), json=request_params, headers=headers)
+        if response_utils.response_error(response):
+            print 'Creating log failed, status code: %d' % response.status_code
+            exit(1)
+        elif response.status_code == 201:
+            pretty_print_string_as_json(response.text)
+
+    except requests.exceptions.RequestException as error:
+        print error
+        exit(1)
+
+
+def delete_log(log_id):
+    """
+    Delete a log with the provided log ID.
+    """
+    url = _url() + '/' + log_id
+    headers = api_utils.generate_headers('rw')
+
+    try:
+        response = requests.delete(url, headers=headers)
+        if response_utils.response_error(response):
+            print 'Delete log failed, status code: %d' % response.status_code
+            exit(1)
+        elif response.status_code == 204:
+            print 'Deleted log with id: %s' % log_id
+    except requests.exceptions.RequestException as error:
+        print error
+        exit(1)
+
+
+def patch_log(log_id, url, headers, params):
```

Comment:
At the moment, this is very much in the style of the rest of the project. I think if we want to change it, it makes sense to change it elsewhere in the project too.

I don't mind doing this work, but given that we're OOO tomorrow, I think this might make the card bleed into next sprint a bit.

Thoughts?